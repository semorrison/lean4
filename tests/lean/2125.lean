open Classical

noncomputable def f (α : Type) : Bool :=
  Nonempty α

-- The following inductive is unsound:
inductive C : Bool → Type
  -- Must be rejected because `C` occurs inside an index.
  | c : C (f (C false))
