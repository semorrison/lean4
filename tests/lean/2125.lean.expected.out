2125.lean:7:0-9:23: error: (kernel) invalid return type for 'C.c'
